react/week4/my-app/src/Context.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from "react";
+
+const Context = React.createContext();
+
+function ContextProvider({ children }) {
+  const [userInput, setUserInput] = useState("");
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  useEffect(() => {
+    if (!userInput || error) {
+      setLoading(false);
+    } else {
+      setLoading(true);
+      fetchGitUsers(userInput);
+    }
+  }, [userInput, error]);
+  const fetchGitUsers = async (input) => {
+    try {
+      await fetch(`https://api.github.com/search/users?q=${input}`)
+        .then((response) => {
+          if (!response.ok) {
+            throw Error("Could not fetch the data ");
+          } else {
+            return response.json();
+          }
+        })
+        .then((data) => {
+          setData(data.items);
+          //console.log(data);
+          setLoading(false);
+        });
+    } catch (error) {
+      setError(error.message);
+    }
+  };
+
+  return (
+    <Context.Provider
+      value={{
+        userInput,
+        setUserInput,
+        data,
+        setData,
+        loading,
+        setLoading,
+        error,
+        setError,
```

Comment:
Do you need to put all these values in to the context? Ideally, we should only put things into the context that we might need from other components and something like `setLoading` for instance is only used in this `ContextProvider` component.